pkg/signup/signup_service_test.go

```diff
@@ -0,0 +1,107 @@
+package signup_test
+
+import (
+	"context"
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/signup"
+	testutils "github.com/codeready-toolchain/registration-service/test"
+	"github.com/codeready-toolchain/registration-service/test/fake"
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"testing"
+)
+
+const (
+	TestNamespace = "test-namespace-123"
+)
+
+type TestSignupServiceSuite struct {
+	testutils.UnitTestSuite
+}
+
+func TestRunSignupServiceSuite(t *testing.T) {
+	suite.Run(t, &TestSignupServiceSuite{testutils.UnitTestSuite{}})
+}
+
+func (s *TestSignupServiceSuite) TestNewSignupService() {
+	// Simply test creation of the service, which should fail as the kubernetes env variables are not set
+	_, err := signup.NewSignupService(configuration.CreateEmptyRegistry())
+	require.Error(s.T(), err)
+}
+
+func (s *TestSignupServiceSuite) TestCreateUserSignup() {
+	svc, fake := newSignupServiceWithFakeClient()
```

Comment:
we should probably try to use given/when/then to keep this consistent with the other repositories. we didn't do this in our other tests :-( guess i'll open a PR to add that tmrw if we decide we should add it here..

Reply:
I'm not a fan of this convention, it doesn't always fit in when a test is modelling complex workflows, so I'd prefer not to make it universally enforcable.